extension/httpfs/s3fs.cpp

```diff
@@ -487,10 +487,64 @@ std::shared_ptr<S3WriteBuffer> S3FileSystem::GetBuffer(S3FileHandle &file_handle
 	return new_write_buffer;
 }
 
-ParsedS3Url S3FileSystem::S3UrlParse(string url, const S3AuthParams &params) {
+void S3FileSystem::readQueryParams(const string &url_query_param, S3AuthParams &params) {
+	if (url_query_param.empty())
+		return;
+
+	duckdb_httplib_openssl::Params query_params;
+	duckdb_httplib_openssl::detail::parse_query_text(url_query_param, query_params);
+
+	auto found_param = query_params.find("s3_region");
+	if (found_param != query_params.end()) {
+		params.region = found_param->second;
+		query_params.erase(found_param);
+	}
+	found_param = query_params.find("s3_access_key_id");
+	if (found_param != query_params.end()) {
+		params.access_key_id = found_param->second;
+		query_params.erase(found_param);
+	}
```

Comment:
Maybe add a method for all of these? Something like
```c++
params.region = GetQueryParam("s3_region", query_params);
```

Reply:
Probably as a reference, as to not alter the param if it wasn't found :)

Reply:
Thank you, I've altered it!